core/src/mindustry/world/blocks/defense/ShockMine.java

```diff
@@ -37,20 +40,30 @@ public void drawTeam(){
         @Override
         public void draw(){
             super.draw();
-            Draw.color(team.color);
-            Draw.alpha(0.22f);
-            Fill.rect(x, y, 2f, 2f);
-            Draw.color();
+            if(isActive() || progress < 20f){
+                Draw.color(team.color);
+                Draw.alpha(0.5f);
+                Fill.rect(x, y, 2f, 2f);
+                Draw.color();
+            }
+            if(!isActive()){
+                progress += delta();
+                progress %= blinkDelay;
+            }
         }
 
         @Override
         public void unitOn(Unit unit){
-            if(enabled && unit.team != team && timer(timerDamage, cooldown)){
+            if(enabled && unit.team != team && timer(timerDamage, cooldown) && isActive()){
```

Comment:
maybe put `isActive()` after `enabled`? so it doesn't need to execute more height statements.